packages/search/src/users/users.ts

```diff
@@ -0,0 +1,171 @@
+/* Copyright (c) 2021 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import { getProp } from "@esri/hub-common";
+import { UserSession } from "@esri/arcgis-rest-auth";
+
+import { GraphQLClient } from "graphql-request";
+import {
+  createSessionMutation,
+  userSelfQuery,
+  userSearchQuery
+} from "./queries";
+
+import { IUser } from "@esri/arcgis-rest-types";
+import {
+  base64,
+  IDateRange,
+  SortDirection,
+  ICursorSearchResults
+} from "../types";
+
+export enum UserSortableField {
+  USERNAME = "USERNAME",
+  LAST_HUB_SESSION = "LAST_HUB_SESSION"
+}
+
+interface ISortingOption {
+  field?: UserSortableField;
+  sortDirection?: SortDirection;
+}
+
+/**
+ * Follows Edge -> Node schema yielded by GraphQL requests
+ */
+interface IEdge {
+  node: IHubUser;
+  cursor?: base64;
+}
+
+/**
+ * Interface that derives directly from the User type in Hub Profiles API
+ */
+export interface IHubUser extends IUser {
+  followedInitiatives?: any[];
+  groups?: any[];
+  teams?: any[];
+  registeredEvents?: any[];
+  lastHubSession?: number;
+  visitsLast30Days?: number;
+  visitsLast60Days?: number;
+}
```

Comment:
I would love to have a `thumbnailUrl` that is a fully qualified url to the user's thumbnail. We do this on `IHubContent` via `IHubResource`.

Reply:
This should be implemented in the hub-profiles I think cc @trescube

Reply:
including @thomas-hervey (he's the PE and all feature requests should go thru him).